index.html

```diff
@@ -1603,6 +1622,348 @@ <h2>
+    </li>
+    <li>
+      If |virtualPressureSourceType| is not a [=string=], return [=error=] with [=error code|WebDriver error code=] [=invalid argument=].
+    </li>
+    <li>
+      Let |sampleInterval| be the result of invoking
+      <a data-cite="!WEBDRIVER2#dfn-getting-properties">get a property</a> "sampleInterval" from |parameters|.
+    </li>
+    <li>
+      If |sampleInterval| is not an Number, return [=error=] with [=error code|WebDriver error code=] [=invalid argument=].
+    </li>
+    <li>
+      Let |samples| be the result of invoking
+      <a data-cite="!WEBDRIVER2#dfn-getting-properties">get a property</a> "samples" from |parameters|.
+    </li>
+    <li>
+      If |samples| is not an Array, return [=error=] with [=error code|WebDriver error code=] [=invalid argument=].
+    </li>
+    <li>
+      For each |sample| in |samples|:
+      <ol>
+        <li>
+          If |sample| is not a [=string=] and a valid {{PressureState}}, return [=error=] with [=error code|WebDriver error code=] [=invalid argument=].
+        </li>
+        <li>
+          [=queue/Enqueue=] |sample| to |virtualPressureSource|'s [=pending samples=].
+        </li>
+      </ol>
+    </li>
+    <li>
+      Let |topLevelTraversable| be the <a data-cite="webdriver2/#dfn-current-browsing-context">current browsing
+      context</a>'s [=browsing context/top-level traversable=].
+    </li>
+    <li>
+      Let |topLevelVirtualPressureSourceMapping| be the |topLevelTraversable|'s [=virtual pressure source mapping=].
+    </li>
+    <li>
+      If |topLevelVirtualPressureSourceMapping| contains |virtualPressureSourceType|, return [=error=] with
+      [=error code|WebDriver error code=] [=invalid argument=].
+    </li>
+    <li>
+      Let |virtualPressureSource| be a new [=virtual pressure source=].
+    </li>
+    <li>
+      Let |virtualPressureSource|'s [=sample update interval=] be |sampleInterval|.
+    </li>
+    <li>
+      Run the following steps [=in parallel=] at the |sampleInterval| time interval:
+      <ol>
+        <li>
+          Remove |virtualPressureSource|'s [=pending samples=] first item if the queue is not empty.
+        </li>
+      </ol>
+    </li>
+    <li>
+      Set |topLevelVirtualPressureSourceMapping|[|virtualPressureSourceType|] to |virtualPressureSource|.
+    </li>
+    <li>
+      Return [=success=] with data <code>null</code>.
+    </li>
+  </ol>
+  <h4>
+    Delete virtual pressure source
+  </h4>
+  <table class="def">
+    <tr>
+      <th>
+        HTTP Method
+      </th>
+      <th>
+        [=extension command URI Template|URI Template=]
+      </th>
+    </tr>
+    <tr>
+      <td>
+        DELETE
+      </td>
+      <td>
+        /session/{session id}/pressuresource/{type}
+      </td>
+    </tr>
+  </table>
+  <p>
+    This [=extension command=] deletes a given [=source type=] of [=virtual pressure source=] and returns
+    pressure source updates back to hardware.
```

Comment:
What does "return pressure source updates back to hardware" mean? For example, what should happen to an active observer for a given source type that is not supported by the platform once the virtual pressure source is removed?

Reply:
The way it is spec'ed now they just wont receive any updates. But I do need to make sure that observers don't fail if a source is not supported by hardware.